crates/symbolicator/src/services/symbolication/process_minidump.rs

```diff
@@ -130,81 +128,187 @@ impl MinidumpState {
+    #[tracing::instrument(skip_all)]
+    async fn load_cfi_module(&self, module: &(dyn Module + Sync)) -> Option<()> {
+        let id = (
+            module.debug_identifier().unwrap_or_default(),
+            module.base_address(),
+        );
+        if self.cficaches.read().contains_key(&id) {
+            return Some(());
+        }
+
+        if id.0.is_nil() {
+            // save a dummy "missing" CfiModule for this id.
+            let cfi_module = CfiModule {
+                cfi_status: ObjectFileStatus::Missing,
+                ..Default::default()
+            };
+            self.cficaches.write().insert(id, cfi_module);
+            return None;
+        }
+
+        let sources = self.sources.clone();
+        let scope = self.scope.clone();
+
+        let identifier = ObjectId {
+            code_id: module.code_identifier(),
+            code_file: Some(module.code_file().into_owned()),
+            debug_id: module.debug_identifier(),
+            debug_file: module
+                .debug_file()
+                .map(|debug_file| debug_file.into_owned()),
+            object_type: self.object_type,
+        };
+
+        let cache_result = self
+            .cficache_actor
+            .fetch(FetchCfiCache {
+                object_type: self.object_type,
+                identifier,
+                sources,
+                scope,
+            })
+            .bind_hub(Hub::new_from_top(Hub::current()))
+            .await;
+
+        let cfi_module = match cache_result {
+            Ok(cfi_cache) => {
+                let cfi_status = match cfi_cache.status() {
+                    CacheStatus::Positive => ObjectFileStatus::Found,
+                    CacheStatus::Negative => ObjectFileStatus::Missing,
+                    CacheStatus::Malformed(details) => {
+                        let err = CfiCacheError::ObjectParsing(ObjectError::Malformed);
                         let stderr: &dyn std::error::Error = &err;
-                        tracing::error!(stderr, "Error while loading cficache");
-                    })
-                    .ok()?;
+                        tracing::warn!(stderr, "Error while parsing cficache: {}", details);
+                        ObjectFileStatus::from(&err)
+                    }
+                    // If the cache entry is for a cache specific error, it must be
+                    // from a previous cficache conversion attempt.
+                    CacheStatus::CacheSpecificError(details) => {
+                        let err = CfiCacheError::ObjectParsing(ObjectError::Malformed);
+                        tracing::warn!("Cached error from parsing cficache: {}", details);
+                        ObjectFileStatus::from(&err)
+                    }
+                };
+                let symbol_file = match cfi_cache.status() {
+                    CacheStatus::Positive => load_symbol_file(cfi_cache.path()).ok(),
```

Comment:
`load_symbol_file()` is a sync function with a bunch of blocking file I/O with some computations, being called from inside an async function.